File: src/main/java/com/hawcore/spark/java/test1/BroadcastVariable.java
package com.hawcore.spark.java.test1;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.api.java.function.VoidFunction;
import org.apache.spark.broadcast.Broadcast;

import java.util.Arrays;
import java.util.List;

/**
 * @author xn025665
 * @date Create on 2019/5/27 14:13
 */
public class BroadcastVariable {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf().setAppName("BroadcastVariable").setMaster("local");
        JavaSparkContext sc = new JavaSparkContext(conf);

        int factor = 3;
        Broadcast<Integer> broadcast = sc.broadcast(3);


        List<Integer> numberList = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
        JavaRDD<Integer> numbers= sc.parallelize(numberList);
        numbers.map(new Function<Integer, Integer>() {
            @Override
            public Integer call(Integer integer) throws Exception {
                return integer * broadcast.value();
            }
        }).foreach(new VoidFunction<Integer>() {
            @Override
            public void call(Integer integer) throws Exception {
                System.out.println(integer);
            }
        });



        sc.close();
    }
}

File: src/main/java/com/hawcore/spark/scala/test3/BroadcastVariable.scala
package com.hawcore.spark.scala.test3

import org.apache.spark.{SparkConf, SparkContext}

/**
  * @author xn025665   
  * @date Create on 2019/5/27 14:20 
  */
object BroadcastVariable {

  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setAppName("LineCount").setMaster("local")

    val sc = new SparkContext(conf)

    val factor = 3;

    val broadcast = sc.broadcast(factor);

    val numberList: Array[Int] = Array(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)


    numberList.map(_ * factor).foreach(println(_))
    numberList.map(_ * broadcast.value).foreach(println(_))
  }
}
